src/lib.rs

```diff
@@ -177,24 +179,10 @@ impl MemtestRunner {
         let mut timed_out = false;
 
         for test_kind in &self.test_kinds {
-            let test = match test_kind {
-                MemtestKind::OwnAddressBasic => memtest::test_own_address_basic,
-                MemtestKind::OwnAddressRepeat => memtest::test_own_address_repeat,
-                MemtestKind::RandomVal => memtest::test_random_val,
-                MemtestKind::Xor => memtest::test_xor,
-                MemtestKind::Sub => memtest::test_sub,
-                MemtestKind::Mul => memtest::test_mul,
-                MemtestKind::Div => memtest::test_div,
-                MemtestKind::Or => memtest::test_or,
-                MemtestKind::And => memtest::test_and,
-                MemtestKind::SeqInc => memtest::test_seq_inc,
-                MemtestKind::SolidBits => memtest::test_solid_bits,
-                MemtestKind::Checkerboard => memtest::test_checkerboard,
-                MemtestKind::BlockSeq => memtest::test_block_seq,
-            };
+            let test = test_kind.to_fn();
```

Comment:
Hmm... I'm finding that this is getting a bit awkward now. Perhaps it would be better, instead, to move this `match()` logic to a block like this in `memtest.rs`:

```Rust
impl MemtestKind {
    pub fn run<O: TestObserver>(&self, memory: &mut usize, observer: O) -> MemtestResult<T> {
        match self {
            // ...
        }
    }
}
```

And then just have the code in `lib.rs` call it through `test_kind.run(...)`.